src/Raven.Server/Documents/Handlers/IndexHandler.cs

```diff
@@ -652,6 +660,19 @@ public Task Terms()
             }
         }
 
+        private string GetIndexNameFromCollectionAndField(string field, DocumentsOperationContext context)
+        {
+            var collection = GetStringQueryString("collection", false);
+            if (string.IsNullOrEmpty(collection))
+                ThrowRequiredMember("name");
+            var query = new IndexQueryServerSide(new QueryMetadata(collection, field));
```

Comment:
Maybe pass the the 'old' constructor `from <collection> select <field>` and parse it using the same flow?

Reply:
I can change it, yes. But it would do more work than required.

Reply:
True, but from the other side, at some point we will add additional logic in first ctor, and 2nd one will be omitted. We can leave as is, but I thought that the change is not that complex?

Reply:
Done, the query won't be cached.